smart-on-fhir-app/src/js/oauth.js

```diff
@@ -0,0 +1,101 @@
+/* 
+  This global variable stores the application token
+  The schema for application token is
+  {
+    "access_token" : "access_token_value",
+    "expires_in"  : "validity of the token",
+    "onSetTime" : "time of token creation"
+  } 
+*/
+var application_token;
+
+// Function to perform the OAuth2.0 authorization
+// It first checks whether application token is present in the local storage
+// If present it checks whether the token has timedout or not
+// If it's still valid, then it's reused.
+// In all other cases, a new token is fetched.
+function oauth() {
+  // Get the token from local storage
+  application_token = window.localStorage.getItem("application_token");
+
+  if (application_token == null) {
+    // If token is not found
+    fetchNewAccessToken();
+  } else {
+    application_token = JSON.parse(application_token);
+    // Check whether the token has timed-out
+    if (checkTimeout()) {
+      fetchNewAccessToken();
+    }
```

Comment:
here return the token in the form
```
{
   "status": "success",
   "token": "thetoken",
   "message": "hit/miss"
}
```
message will be hit if you get access token from storage, it will be miss if the token was expired/not-present and you fetched a new token

if some error occurs:
```
{
   "status": "error",
   "token": null,
   "message": "actual description of the error"
}
```